File: client/src/app/_models/loan-payment-table.ts
export interface LoanPaymentTable {
    id: number;
    totalNumPayments: number;
    rows: LoanPaymentRow[];
}

export interface LoanPaymentRow {
    id : number;
    paymentNumber: number;
    monthlyInstallment: number;
    principalPayment: number;
    interestPayment: number;
    remainingBalance: number;
}

File: API/Entities/LoanPaymentTable.cs
public class LoanPaymentTable
{   
    public int Id { get; set; }
    public int TotalNumPayments { get; set; }
    public List<LoanPaymentRow> Rows { get; set; }
    public LoanPaymentTable()
    {
        Rows = new List<LoanPaymentRow>();
    }
}
